Update tutorials for Codespaces (#426)
nf-core/airrflow: docs/usage/single_cell_tutorial.md

## Pre-requisites

You can run this tutorial using the Github Codespaces platform. Codespaces already has Nextflow and Singularity pre-installed, and it can automatically be used for every nf-core repository. To create a Codespace instance for nf-core/airrflow, first click on the button labelled `Code` at the top of [nf-core/airrflow repository](https://github.com/nf-core/airrflow).

> In the dropdown menu, go to the `Codespaces` tab. You can create a basic "2-core" Codespace by clicking the `+` icon. However, as more CPUs and memories are needed for nf-core/airrflow task, you need to press the `...` sign and choose `+ New with options...`. 

Reviewer: ```suggestion
In the dropdown menu, go to the `Codespaces` tab. After clicking on the `...` sign, and the `+ New with options...` button.
```

> [!NOTE]
> If you want to run this tutorial on your local machine, please refer to [this page](https://nf-co.re/docs/usage/installation) on how to set up Nextflow and a container engine needed to run this pipeline. At the moment, nf-core/airrflow does NOT support using conda virtual environments for dependency management, only containers are supported. Make sure to [test your setup](https://nf-co.re/docs/usage/introduction#how-to-run-a-pipeline) before running the workflow on actual data. To install Docker, follow the [instructions](https://docs.docker.com/engine/install/). After installation Docker on Linux, don't forget to check the [post-installation steps](https://docs.docker.com/engine/install/linux-postinstall/).

## Testing the pipeline with built-in tests

Once you have set up Nextflow and container (Docker or Singularity) for your local machine or Codespace environment, test nf-core/airrflow with the built-in test data.

```bash
nextflow run nf-core/airrflow -r 4.3.1 -profile test,docker --outdir test_results
```

> Change the `docker` profile to `singularity` if you use Codespace since The docker profile currently does not work in Codespaces. 

Reviewer: ```suggestion
Change the `docker` profile to `singularity` if you use Codespaces since Docker currently cannot be used in Codespaces. 
```

Details on the required columns of a samplesheet are available [here](https://nf-co.re/airrflow/usage#assembled-input-samplesheet-bulk-or-single-cell-sequencing).

The resource configuration file sets the compute infrastructure maximum available number of CPUs, RAM memory and running time. This will ensure that no pipeline process requests more resources than available in the compute infrastructure where the pipeline is running. The resource config should be provided with the `-c` option. In this example we set the maximum RAM memory to 16GB, we restrict the pipeline to use 4 CPUs and to run for a maximum of 24 hours.

```json title="resource.config"
process {
    resourceLimits = [ memory: 16.GB, time: 24.h, cpus: 4 ]

> }

Reviewer: ```suggestion
}
apptainer{
  cacheDir = "/workspaces/airrflow/singularity_cache"
}
```
